torch/distributed/pipelining/schedules.py

```diff
@@ -955,3 +956,151 @@ def backward_stage_index(step):
         for _ in range(self.pp_group_size - rank - 1):
             rank_ops.append(None)
         return rank_ops
+
+
+class ScheduleFlexibleInterleaved1F1B(PipelineScheduleMulti):
```

Comment:
We can add this one as one of the schedules to test with varying number of microbatches, stages, and group_size here: https://github.com/pytorch/pytorch/blob/main/test/distributed/pipelining/test_schedule.py#L556. Just have to add it to the `@parametrize` list